main.tf

```diff
@@ -1,33 +1,207 @@
-# Get object aws_vpc by vpc_id
-data "aws_vpc" "default" {
-  count = local.enabled ? 1 : 0
-  id    = var.vpc_id
+# Process all the optional inputs into a fully fleshed out configuration
+# the rest of the code can consume.
+
+locals {
+  enabled = module.this.enabled && (var.public_subnets_enabled || var.private_subnets_enabled) && (var.ipv4_enabled || var.ipv6_enabled)
+
+  # We are going to reference `enabled` a *lot*, so abbreviate it
+  e = local.enabled
```

Comment:
I will have to say that for consistency with all other modules we should not abbreviate it

Reply:
It is available in unabbreviated form as well, and the unabbreviated form is used everywhere outside of `locals {}`, so I feel that is consistent enough.

Reply:
Yeah, I don't see an issue with the `local.e` usage within the `locals {}` as it is. To me this is clean and easy to track, there is no inconsistency as I see as it is only used for the complex `locals` and everything outside of it is using the unabbreviated form.